### PR TITLE
implemented the flow for creating a new ssp

### PR DESCRIPTION
**Pull request by dominicmeddick**

### CODE REVIEW

File: api/handler/ssp.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// @Description 	Create a catalog with the given title
+// @Tags 			Catalog
+// @Accept  		json
+// @Produce  		json
+// @Param   		catalog body createCatalogRequest true "Catalog to add"
+// @Success 		201 {object} catalogIdResponse
+// @Failure 		401 {object} api.Error
+// @Failure 		422 {object} api.Error
+// @Failure 		500 {object} api.Error
+// @Router 			/api/catalog [post]
```
Comment (onselakin) [inline api/handler/ssp.go:34]: Can you fix and check the Swagger definition? The route should be something like /api/ssp. Request and response types should also be fixed.

File: api/handler/response.go [hunk 2]
```diff
@@ -17,3 +17,12 @@ type planIdResponse struct {
 	// Example: "456def"
 	Id string `json:"id"`
 }
+
+// sspIdResponse is a struct that holds the ID of a SSP.
+// swagger:model
+type sspIdResponse struct {
```
Comment (onselakin) [inline api/handler/response.go:23]: You can remove this one and use idResponse struct

File: api/handler/ssp.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+type SSPHandler struct {
+	service *service.SSPService
+}
+
+func NewSSPHandler() *SSPHandler {
+	return &SSPHandler{}
+}
+
+func (h *SSPHandler) Register(api *echo.Group) {
+	api.POST("/systemsecurityplan", h.CreateSSP)
```
Comment (onselakin) [inline api/handler/ssp.go:20]: Let's shorten this to /ssp. See my next comment about the routes.

File: api/handler/request.go [hunk 4]
```diff
@@ -25,6 +26,19 @@ func (r *createCatalogRequest) bind(ctx echo.Context, c *domain.Catalog) error {
 	return nil
 }
 
+type createSSPRequest struct {
```
Comment (onselakin) [inline api/handler/request.go:29]: Can you add a comment to describe what this does, like the other request types?

**Review by onselakin — CHANGES_REQUESTED** [review]

File: api/handler/response.go (hunk 2 above)
Comment (dominicmeddick) [reply api/handler/response.go:23]: There is no idResponse struct defined. Did you mean to create idResponse struct to replace all 3 (catalog,plan,ssp) defined here currently because they all look the same?

Comment (onselakin) [reply api/handler/response.go:23]: Your branch is 17 commits behind main. You should update 😊

**Review by onselakin — APPROVED** [review]
